src/index.ts

```diff
@@ -136,7 +136,7 @@ export interface UploadConfig {
   userProject?: string;
 }
 
-export interface ConfigMetadata {
+export interface ConfigMetadata extends Object {
```

Comment:
what does extending `Object` do!?  Is this a class....?

Reply:
Lol, I don't know! I just needed to let arbitrary properties exist, and this pleased the linter. Is there a smarter way?

Reply:
I would suggest not extending Object.  If this is truly dynamic (supports any property based on user data), then the `any` type is what we should be using.

Reply:
Gave it another shot, please take a look!